From 03749d9835d219f6d3ae5c31741007fd84a96b3b Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Wed, 15 Jan 2014 16:29:08 -0800
Subject: [PATCH 222/793] drm: xilinx: drv: Add bpp to
 xilinx_video_format_desc

Add bpp(bits per pixel) information to xilinx_video_format_desc.
This information can be used when an initial framebuffer is created.

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_drv.c |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_drv.c b/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
index d528004..0070a37 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
@@ -54,20 +54,22 @@ struct xilinx_drm_private {
 /**
  * struct xilinx_video_format_desc - Xilinx Video IP video format description
  * @name: Xilinx video format name
+ * @bpp: bits per pixel
  * @xilinx_format: xilinx format code
  * @drm_format: drm format code
  */
 struct xilinx_video_format_desc {
 	const char *name;
+	unsigned int bpp;
 	unsigned int xilinx_format;
 	uint32_t drm_format;
 };
 
 static const struct xilinx_video_format_desc xilinx_video_formats[] = {
-	{ "yuv422", XILINX_VIDEO_FORMAT_YUV422, DRM_FORMAT_YUV422 },
-	{ "yuv444", XILINX_VIDEO_FORMAT_YUV444, DRM_FORMAT_YUV444 },
-	{ "xrgb888", XILINX_VIDEO_FORMAT_RGB, DRM_FORMAT_XRGB8888 },
-	{ "yuv420", XILINX_VIDEO_FORMAT_YUV420, DRM_FORMAT_YUV420 },
+	{ "yuv422", 16, XILINX_VIDEO_FORMAT_YUV422, DRM_FORMAT_YUV422 },
+	{ "yuv444", 24, XILINX_VIDEO_FORMAT_YUV444, DRM_FORMAT_YUV444 },
+	{ "xrgb888", 32, XILINX_VIDEO_FORMAT_RGB, DRM_FORMAT_XRGB8888 },
+	{ "yuv420", 16, XILINX_VIDEO_FORMAT_YUV420, DRM_FORMAT_YUV420 },
 };
 
 /* create a fb */
-- 
1.7.5.4

